deploy/charts/secret-manager/templates/deployment.yaml

```diff
@@ -22,7 +22,9 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-      serviceAccountName: {{ include "secret-manager.serviceAccountName" . }}
+      {{- if or .Values.serviceAccount.create .Values.serviceAccount.name }}
+      serviceAccountName: {{ .Values.serviceAccount.name | default (include "secret-manager.serviceAccountName" .) }}
```

Comment:
I'm ok with if statement wrapping the field, but I would prefer to leave the value as `{{ include "secret-manager.serviceAccountName" . }}`. Using the template function makes it much more simple when referencing each time in Rolebindings and other resources.